File: C/Searching/linear_search.c
#include<stdio.h>
#include<stdlib.h>

void linear_search(int A[], int n, int value) 
{
    int i, flag = 0;
    for (i=0;i<n;i++)
    {
        if (A[i] == value)
        {
            flag = 1;
            printf("\nData found at position %d", i+1);
            break;
        }
    }

    if (flag == 0)
    {
        printf("\nData is not found");
    }
}

int main()
{
    int n, i, query;
    int *ptr;
    printf("\nEnter how many elements?");
    scanf("%d", &n);
    ptr = (int*)malloc(n*sizeof(int));
    for (i = 0;i<n;i++)
    {
        scanf("%d", &ptr[i]);

    }
    printf("\nThe data is ");
    for (i = 0;i<n;i++)
    {
       printf("\n\t%d", ptr[i]);
    }
    printf("\nEnter the search value:");
    scanf("%d", &query);

    linear_search(ptr,n,query);
    return 0;

}

File: Python/Searching/linear_search.py
import random
import time


class Data:
    def __init__(self):
        self.items = []
        self.num = None

    def get(self):
        self.num = int(input("Enter how many number:"))
        for i in range(self.num):
            item = int(input("Enter data:"))
            self.insert(item)


    def insert(self, item):
        self.items.append(item)

   

    def random(self):
        self.items = [x+1 for x in range(self.num)]
        random.seed(time.time())
        random.shuffle(self.items)

    def display(self):
        print(f'{self.items}')

    def search(self):
        val = int(input("Enter the data you want to search:"))
        flag = False
        if self.items:
            for index, value in enumerate(self.items):
                if (value == val):
                    flag = True
                    print("Data is found at ",index+1)
                    break

            if flag == False:
                print("Search not found.")

            





    

if __name__ == "__main__":    
    data = Data()
    data.get()
    data.display()
    data.search()
    
        
